### ISSUE TITLE
tab:set_zoomed not working when active pane is manually set

### ISSUE THREAD

**Issue by SpyMachine** [wez/wezterm#5928]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20240811-112835-a0a415d1

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I'm trying to setup a workspace initializer where I have my editor and a terminal that is hidden by default. I.e. I want to zoom my editor by default. However the configuration (see below) doesn't work. The `editor_pane` is not zoomed. Interestingly, if I comment out `editor_pane:activate()` it does zoom the terminal pane which I'd expect so I think perhaps there's an issue with the zoom when the pane is "manually" activated.

### To Reproduce

For simplicity, I was able to reproduce the issue on "gui-startup" just to make it easy to test. So just copying that gui-startup event handler should show the issue.

### Configuration

```
wezterm.on("gui-startup", function()
	local _, _, window = wezterm.mux.spawn_window({})
	window:gui_window():maximize()
	local editor_tab = window:active_tab()
	editor_tab:set_title("editor")
	local editor_pane = editor_tab:active_pane()
	editor_pane:send_text("nvim -c ':Telescope find_files' .\n")
	local _terminal_pane = editor_pane:split({ direction = "Bottom", size = 0.15 })
	editor_pane:activate()
	editor_tab:set_zoomed(true)
end)
```

### Expected Behavior

I expected my "editor_pane" to be zoomed using the above configuration.

**Comment by SpyMachine** [comment]
oh interesting.... printing out `tab:pane_with_info`, wezterm seems to think the pane is zoomed but it's not.

```
20:34:58.272  INFO   logging > lua: [
    {
        "height": 20,
        "index": 0,
        "is_active": true,
        "is_zoomed": true,
        "left": 0,
        "pane": MuxPane(pane_id:0, pid:85366),
        "pixel_height": 680,
        "pixel_width": 1280,
        "top": 0,
        "width": 80,
    },
    {
        "height": 3,
        "index": 1,
        "is_active": false,
        "is_zoomed": false,
        "left": 0,
        "pane": MuxPane(pane_id:1, pid:85366),
        "pixel_height": 102,
        "pixel_width": 1280,
        "top": 21,
        "width": 80,
    },
]
```

**Comment by SpyMachine** [comment]
Okay, I did some debugging last night and I think I see what is happening. I think there's a race condition between the [event handler for PaneFocused](https://github.com/wez/wezterm/blob/0ac1e948c5b4fb1d1dee532eba36c4669a64774e/wezterm-gui/src/frontend.rs#L80-L88) and where I'm calling:

```
editor_pane:activate()
editor_tab:set_zoomed(true)
```

It seems like set_zoomed is working initially, however then the event handler runs for activate runs after and eventually calls ["toggle_zoom"](https://github.com/wez/wezterm/blob/0ac1e948c5b4fb1d1dee532eba36c4669a64774e/mux/src/tab.rs#L1755). 

I ran out of time last night digging into the race condition aspect but I did find a potential alternative way to solve my problem. I saw the configuration option used here for ["unzoom_on_switched_pane"](https://github.com/wez/wezterm/blob/0ac1e948c5b4fb1d1dee532eba36c4669a64774e/mux/src/tab.rs#L1752-L1754) and realized that in my case the active pane *is* the zoomed pane. So I wondered if we could also early return in the case where they are the same as it seems surprising that focusing the zoomed pane would unzoom... it. I.e.

```
-            if !configuration().unzoom_on_switch_pane {
+            if !configuration().unzoom_on_switch_pane || is_pane(pane, &self.zoomed.as_ref()) {
                 return;
             }
```

I'm obviously oblivious to what knock on effects this could have though, so feel free to tell me that's a bad idea :)
